### PR TITLE
replace orderedset with ordered_set (notify-admin-1137)

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

In pyproject.toml, notifications-utils is set to a rev that is currently in code review, which shows the original orderedset has been completely replaced by ordered_set.

## Security Considerations

N/A

### CODE REVIEW

**Comment by ccostino** [comment]
Noting here that once the `ordered-set` PR for notification-utils is merged, this PR will have to be updated to account for the new utils version bump, which will take care of the dependency updates needed on this side.

File: poetry.lock
Comment (xlorepdarkhelm) [inline poetry.lock:1]: I did a grep on this file and saw `orderedset` still in it. Line 1707.

**Review by xlorepdarkhelm — CHANGES_REQUESTED** [review]
Looks pretty good, I found one instance of `orderedset` still existing, inside the poetry.lock file, as noted. Not sure if that's a big deal or not, I'm not used to seeing the pyproject.toml data inside poetry.lock.

File: poetry.lock
Comment (terrazoon) [reply poetry.lock:1]: But I need to merge this PR, or Samatha will be blocked (or have to redo the work) when she tries to test her changes on notifications-util.  This is a chicken and egg problem, and I think this PR needs to be merged to break the deadlock.

Comment (terrazoon) [reply poetry.lock:1]: See this PR:  https://github.com/GSA/notifications-utils/pull/113

Samatha can't test PR #113 successfully unless this (my) PR is merged.

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM

**Comment by ccostino** [comment]
I swear I'm not ignoring this PR!  I just caught up with @samathad2023 about the Python updates and they're moving along smoothly, and she's been able to test even without this merged just yet.

There's some coordination to be done with `notification-utils` given that the `ordered_set` update is now rolled into the Python version upgrade work there, and I want to make sure things go in at the right time.  The main reason is that once utils gets that update, it's going to reference Python 3.12, and neither the API nor admin will be ready until they also reference Python 3.12 for their runtime environment.

Furthermore, by setting the `notification-utils` reference to a specific revision like that, we risk either bringing things in at the wrong time, or missing updates when we intend to bring them in.

Once the Python version update PRs are also in and ready for review, I'll work with everyone to figure out the best way to get utils updated and account for this as well.  Hold tight, we're almost there! 🙂

**Pull request closed without merging**